### PR TITLE
[NFC][clang] Update ubsan-trap-merge.c test to show absence of nomerge in non-trap mode

### PR DESCRIPTION
**Pull request by thurstond**
This shows that ubsan handlers do not have nomerge attributes in non-trap mode, even if -ubsan-unique-trap is enabled.

0d15d46362bd6ab5a9a2165805adaab13a7689f4 attaches nomerge but only for trap mode.

### CODE REVIEW

File: clang/test/CodeGen/ubsan-handler-merge.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
+// NOTE: Assertions have mostly been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 5
+// The most important assertion is the attributes at the end of the file, which
+// shows that -ubsan-unique-traps does not attach 'nomerge' to each ubsan handler.
+//
+// RUN: %clang_cc1 -triple x86_64-linux-gnu -emit-llvm -fsanitize=signed-integer-overflow -O3 -mllvm -ubsan-unique-traps %s -o - \
```
Comment (vitalybuka) [inline clang/test/CodeGen/ubsan-handler-merge.c:5]: Have you consider to have these in the same test file as traps?

Comment (vitalybuka) [inline clang/test/CodeGen/ubsan-handler-merge.c:5]: This test is good, but most important is https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#minimal-runtime

Comment (thurstond) [reply clang/test/CodeGen/ubsan-handler-merge.c]: Done

Comment (thurstond) [reply clang/test/CodeGen/ubsan-handler-merge.c]: Done

**Review by vitalybuka — APPROVED** [review]

**Pull request merged**